cloudnetpy/instruments/hatpro.py

```diff
@@ -62,73 +72,65 @@ def hatpro2nc(
-    for file in files:
+def _get_hatpro_objects(directory: Path, expected_date: Union[str, None]) -> List[List[HatproBin]]:
+    objects = defaultdict(list)
+    for filename in directory.iterdir():
         try:
-            obj = HatproBin(file)
-            obj.screen_bad_profiles()
+            obj: HatproBin
+            extension = filename.suffix.upper()
+            if extension == ".LWP":
+                obj = HatproBinLwp(filename)
+            elif extension == ".IWV":
+                obj = HatproBinIwv(filename)
+            else:
+                continue
+            obj.mask_bad_profiles()
             if expected_date is not None:
                 obj = _validate_date(obj, expected_date)
+            objects[filename.stem].append(obj)
         except (TypeError, ValueError) as err:
             logging.warning(err)
             continue
-        objects.append(obj)
-        valid_files.append(file)
-    return objects, valid_files
+    return [value for key, value in sorted(objects.items())]
 
 
-def _validate_date(obj, expected_date: str):
-    if obj.header["_time_reference"] == 0:
-        raise ValueError("Ignoring a file (can not validate non-UTC dates)")
+def _validate_date(obj: HatproBin, expected_date: str):
+    if obj.header["_time_reference"] != 1:
+        raise ValueError(f"Ignoring file '{obj.filename}' (can not validate non-UTC dates)")
     inds = []
     for ind, timestamp in enumerate(obj.data["time"][:]):
         date = "-".join(utils.seconds2date(timestamp)[:3])
         if date == expected_date:
             inds.append(ind)
     if not inds:
-        raise ValueError("Ignoring a file (time stamps not what expected)")
+        raise ValueError(f"Ignoring file '{obj.filename}' (timestamps not what expected)")
     for key in obj.data.keys():
         obj.data[key] = obj.data[key][inds]
     return obj
 
 
-DEFINITIONS = {
-    "retrieval_method": (
-        "\n"
-        "Value 0: Linear Regression\n"
-        "Value 1: Quadratic Regression\n"
-        "Value 2: Neural Network"
-    ),
-    "quality_flag": (
-        "\n"
-        "Bit 0: Rain information (0=no rain, 1=raining)\n"
-        "Bit 1/2: Quality level (0=Not evaluated, 1=high, 2=medium, 3=low)\n"
-        "Bit 3/4: Reason for reduced quality"
-    ),
-}
-
-ATTRIBUTES = {
-    "file_code": MetaData(long_name="File code", comment="RPG HATPRO software version.", units="1"),
-    "program_number": MetaData(
-        long_name="Program number",
-    ),
-    "retrieval_method": MetaData(
-        long_name="Retrieval method", definition=DEFINITIONS["retrieval_method"], units="1"
-    ),
-    "quality_flag": MetaData(
-        long_name="Quality flag",
-        definition=DEFINITIONS["quality_flag"],
-        units="1",
-        comment="Quality information as an 8 bit array. See RPG HATPRO manual for more\n"
-        "information.",
-    ),
-}
+class _HatproBinCombined(HatproBin):
+    # pylint: disable=abstract-method, super-init-not-called
```

Comment:
Tarvitseeko olla aliluokka? Voisi ehkä olla ihan omana luokkanaan (vaikka `rpg_reader.py` fileessä, jos tulee circular import ongelmia).

Reply:
Näin on selvästi siistimpi!